src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/runtime/VMObjectFactory.java

```diff
@@ -58,8 +58,8 @@ were desired (though the current system is designed to not require
       return c.newInstance(new Object[] { addr });
     }
     catch (java.lang.reflect.InvocationTargetException ite) {
-        if (ite.getTargetException() instanceof RuntimeException) {
-            throw (RuntimeException)ite.getTargetException();
+        if (ite.getCause() instanceof RuntimeException) {
+            throw (RuntimeException)ite.getCause();
```

Comment:
This might be a place to use the new instanceof pattern form:
`if (ite.getCause() instanceof RuntimeException rex)
        throw rex.getCause();
`